### ISSUE TITLE
Alcatel-Lucent(Nokia) router ssh connect timeout

### ISSUE THREAD

**Issue by sshuguan** [CiscoTestAutomation/pyats#40]
I tried to use pyats to connect Alcatel-Lucent(Nokia) router but ssh timeouts. What is going wrong ?

my tb.yaml
```
devices:
    SR1-Services-Leaf2:
        os: sros
        type: sros
        connections:
            cli:
                ip: 135.228.1.225
                protocol: ssh
            netconf:
                ip: 135.228.1.225
                port: 830
        credentials:
            default:
                password: Nokia2018!
                username: admin
```
my script:
```
from genie.testbed import load
testbed = load("./tb.yaml")
dev = testbed.devices["SR1-Services-Leaf2"]
dev.connect(log_stdout = True, via="cli")
```
run log
```
(pyats) [sshanggu@atims1 bell.ca]$ python ./NokiaCLITest.py 
[2020-02-13 15:25:19,427] +++ SR1-Services-Leaf2 logfile /tmp/SR1-Services-Leaf2-cli-20200213T152519425.log +++
[2020-02-13 15:25:19,427] +++ Unicon plugin sros +++

admin@135.228.1.225's password: 
[2020-02-13 15:25:19,496] +++ connection to spawn: ssh -l admin 135.228.1.225, id: 140098714212000 +++
[2020-02-13 15:25:19,497] connection to SR1-Services-Leaf2


[]
A:admin@SR1-Services-Leaf2# 
[2020-02-13 15:25:19,709] +++ initializing handle +++
//
INFO: CLI #2051: Switching to the classic CLI engine
*A:SR1-Services-Leaf2# Traceback (most recent call last):
  File "src/unicon/statemachine/statemachine.py", line 719, in unicon.statemachine.statemachine.StateMachine.go_to
  File "src/unicon/statemachine/statetransition.py", line 217, in unicon.statemachine.statetransition.StateTransition.do_transitions
  File "src/unicon/eal/dialogs.py", line 430, in unicon.eal.dialogs.Dialog.process
  File "src/unicon/eal/dialog_processor.py", line 300, in unicon.eal.dialog_processor.SimpleDialogProcessor.process
  File "src/unicon/eal/dialog_processor.py", line 259, in unicon.eal.dialog_processor.SimpleDialogProcessor.timeout_handler
unicon.core.errors.TimeoutError: timeout occurred:
              timeout value: 60
              last_command: '//\r'
             pattern: ['^A:SR1-Services-Leaf2(>.*)?#\\s?$']
             buffer:'//\r\nINFO: CLI #2051: Switching to the classic CLI engine\r\n*A:SR1-Services-Leaf2# '

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "src/unicon/bases/connection.py", line 623, in unicon.bases.connection.Connection.connect
  File "src/unicon/bases/routers/connection_provider.py", line 146, in unicon.bases.routers.connection_provider.BaseSingleRpConnectionProvider.connect
  File "/home/sshanggu/pyats/lib/python3.7/site-packages/unicon/plugins/sros/connection_provider.py", line 21, in init_handle
    timeout=con.connection_timeout)
  File "src/unicon/statemachine/statemachine.py", line 722, in unicon.statemachine.statemachine.StateMachine.go_to
unicon.core.errors.StateMachineError: Failed while bringing device to "classiccli" state

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "./NokiaCLITest.py", line 7, in <module>
    dev.connect(log_stdout = True, via="cli")
  File "src/genie/conf/base/device.py", line 599, in genie.conf.base.device.Device.connect
  File "src/pyats/connections/manager.py", line 334, in pyats.connections.manager.ConnectionManager.connect
  File "src/unicon/bases/connection.py", line 629, in unicon.bases.connection.Connection.connect
unicon.core.errors.ConnectionError: failed to connect to SR1-Services-Leaf2
Failed while bringing device to "classiccli" state
(pyats) [sshanggu@atims1 bell.ca]$ 
```

**Comment by tahigash** [comment]
Hi @sshuguan 

The issue was, our Unicon was expecting to see prompt "A:SR1-Services-Leaf2#" after switching to classiscli. but the prompt has '*' at the beginning of line.
Could you please confirm that prompt is expected and what kind of situation we see the '*'? If you can provide the doc url, it would be appreciated. We will try to update our Unicon plugin accordingly.

And if you don't need to change to classiccli, please try below in your testbed yaml.
```
devices:
    SR1-Services-Leaf2:
        os: sros
        type: sros
        connections:
            cli:
                ip: 135.228.1.225
                protocol: ssh
                settings:
                  DEFAULT_CLI_ENGINE: mdcli
            netconf:
                ip: 135.228.1.225
                port: 830
        credentials:
            default:
                password: Nokia2018!
                username: admin

**Comment by sshuguan** [comment]
Hi tahigash,

With regard to "*" character at front of prompt, please refer to:
https://documentation.nokia.com/html/0_add-h-f/93-0070-10-01/7750_SR_OS_System_Basics_Guide/CLI%20Usage.html#2178948
( a “*” appears in the prompt string indicating that the changes have not been saved) 

The existing pattern: 
'^A:SR1-Services-Leaf2(>.*)?#\\s?$'

may I  suggested pattern:
'\\*?**A:SR1-Services-Leaf2(>.*)?#\\s?$'
or
r'^\*?A:SR1-Services-Leaf2(>.*)?#\s?$'

Thanks,
sshanggu

**Comment by tahigash** [comment]
Hi @sshuguan ,

Thanks for the doc. I've opened below pull request in unicon.plugins repo.
https://github.com/CiscoTestAutomation/unicon.plugins/pull/1

The fix will be in next release v20.3 (end of March). or once the PR is merged, the fix can be used by using latest source on GitHub. please let me know if we can close this issue.

**Comment by simingy** [comment]
assuming issue is fixed - please re-open if still there
